src/dash-table/dash/DataTable.js

```diff
@@ -19,1362 +12,22 @@ import LazyLoader from 'dash-table/LazyLoader';
-
-    /**
-     * `derived_virtual_selected_row_ids` represents the IDs of the
-     * `selected_rows` as they appear after filtering and sorting,
-     * across all pages.
-     */
-    derived_virtual_selected_row_ids: PropTypes.arrayOf(
-        PropTypes.oneOfType([PropTypes.string, PropTypes.number])
-    ),
-
-    /**
-     * Object that holds the loading state object coming from dash-renderer
-     */
-    loading_state: PropTypes.shape({
-        /**
-         * Determines if the component is loading or not
-         */
-        is_loading: PropTypes.bool,
-        /**
-         * Holds which property is loading
-         */
-        prop_name: PropTypes.string,
-        /**
-         * Holds the name of the component that is loading
-         */
-        component_name: PropTypes.string
-    }),
-
-    /**
-     * Used to allow user interactions in this component to be persisted when
-     * the component - or the page - is refreshed. If `persisted` is truthy and
-     * hasn't changed from its previous value, any `persisted_props` that the
-     * user has changed while using the app will keep those changes, as long as
-     * the new prop value also matches what was given originally.
-     * Used in conjunction with `persistence_type` and `persisted_props`.
-     */
-    persistence: PropTypes.oneOfType([
-        PropTypes.bool,
-        PropTypes.string,
-        PropTypes.number
-    ]),
+const DataTable = props => (
+    <Suspense fallback={null}>
+        <RealDataTable {...props} />
+    </Suspense>
+);
 
-    /**
-     * Properties whose user interactions will persist after refreshing the
-     * component or the page.
-     */
-    persisted_props: PropTypes.arrayOf(
-        PropTypes.oneOf([
-            'columns.name',
-            'data',
-            'filter_query',
-            'hidden_columns',
-            'selected_columns',
-            'selected_rows',
-            'sort_by'
-        ])
-    ),
+export default DataTable;
 
-    /**
-     * Where persisted user changes will be stored:
-     * memory: only kept in memory, reset on page refresh.
-     * local: window.localStorage, data is kept after the browser quit.
-     * session: window.sessionStorage, data is cleared once the browser quit.
-     */
-    persistence_type: PropTypes.oneOf(['local', 'session', 'memory'])
-};
+const RealDataTable = asyncDecorator(DataTable, () =>
+    LazyLoader.table().then(res => {
+        const realDataTable = res.default;
 
-DataTable.persistenceTransforms = {
-    columns: {
-        name: {
-            extract: propValue => R.pluck('name', propValue),
-            apply: (storedValue, propValue) =>
-                R.zipWith(R.assoc('name'), storedValue, propValue)
-        }
-    }
-};
+        DataTable.defaultProps = realDataTable.defaultProps;
+        DataTable.propTypes = realDataTable.propTypes;
+        DataTable.persistenceTransforms = realDataTable.persistenceTransforms;
```

Comment:
When the real component is loaded, assign the props that matter - this step could be done in the decorator to ensure consistency for components.

Reply:
@Marc-Andre-Rivet this is certainly an interesting idea (and if we go ahead with it I agree doing it at the decorator level would be preferable)... but how do we know the renderer waits for this code before trying to [validate props](https://github.com/plotly/dash/blob/58eb07bf4f2f9d64054fdc65f9a1a17cb6fc355e/dash-renderer/src/TreeContainer.js#L41)?

Reply:
Good question. Made me realize this is not that generalizable as-is. This works for the table because of the derived props. Once the real component is loaded it is guaranteed to call `setProps` which will force the renderer to re-renderer the associated parent TreeContainer, making this work for the general case is possible but more involved and would involve both a decorator update and a tweak to the TreeContainer.

Reply:
One possible generalization: https://github.com/plotly/dash/pull/1372/files